services/snapshotter/client.go

```diff
@@ -0,0 +1,82 @@
+package snapshotter
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/influxdata/influxdb/services/meta"
+	"github.com/influxdata/influxdb/tcp"
+)
+
+// Client provides an API for the snapshotter service.
+type Client struct {
+	host string
+}
+
+// NewClient returns a new *Client.
+func NewClient(host string) *Client {
+	return &Client{host: host}
+}
+
+// MetastoreBackup returns a snapshot of the meta store.
+func (c *Client) MetastoreBackup() (*meta.Data, error) {
+	req := &Request{
+		Type: RequestMetastoreBackup,
+	}
+
+	b, err := c.doRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check the magic.
+	magic := binary.BigEndian.Uint64(b[:8])
+	if magic != BackupMagicHeader {
+		return nil, errors.New("invalid metadata received")
+	}
+	i := 8
+
+	// Size of the meta store bytes.
+	length := int(binary.BigEndian.Uint64(b[i : i+8]))
+	i += 8
+	metaBytes := b[i : i+length]
+	i += int(length)
+
+	// Size of the node.json bytes.
```

Comment:
Ah.  I missed that this was the client.  I thought I was looking at the snapshotter service.